pubsub/src/main/java/com/github/twitch4j/pubsub/TwitchPubSubConnectionPool.java

```diff
@@ -0,0 +1,79 @@
+package com.github.twitch4j.pubsub;
+
+import com.github.twitch4j.common.pool.TwitchModuleConnectionPool;
+import com.github.twitch4j.pubsub.domain.PubSubRequest;
+import lombok.experimental.SuperBuilder;
+import org.apache.commons.lang3.RandomStringUtils;
+
+import java.util.Collection;
+import java.util.stream.StreamSupport;
+
+/**
+ * A pool for PubSub connections to help navigate rate-limits.
+ * <p>
+ * Warning: Passing a {@link java.util.concurrent.ScheduledThreadPoolExecutor} with too small corePoolSize can lead to
+ * connections in the pool not behaving properly. Not specifying an executor allows connections to create their own
+ * at will. If enough connections are made, this could pollute one's runtime environment.
+ */
+@SuperBuilder
+public class TwitchPubSubConnectionPool extends TwitchModuleConnectionPool<TwitchPubSub, PubSubRequest, PubSubSubscription, Class<Void>, TwitchPubSubBuilder> {
+
+    private final String threadPrefix = "twitch4j-pool-" + RandomStringUtils.random(4, true, true) + "-pubsub-";
+
+    @Override
+    public PubSubSubscription subscribe(PubSubRequest pubSubRequest) {
+        final int topics = getTopicCount(pubSubRequest);
+        if (topics <= 0) return null;
+        if (topics > 1) throw new IllegalArgumentException("TwitchPubSubConnectionPool can only handle PubSubRequest's with a single topic subscription");
+        return super.subscribe(pubSubRequest);
```

Comment:
If there was an error in subscription (e.g. BADAUTH), the connection ought to be decremented